include/ua_server.h

```diff
@@ -119,6 +119,12 @@ typedef struct {
     UA_StatusCode (*write)(void *handle, const UA_NodeId nodeid, const UA_Variant *data, const UA_NumericRange *range);
 } UA_DataSource;
 
+typedef struct {
+    void *handle;
+    void (*onRead)(void *handle, const UA_NodeId nodeid,  const UA_Variant *data, const UA_NumericRange *range);
+    void (*onWrite)(void *handle, const UA_NodeId nodeid, const UA_Variant *data, const UA_NumericRange *range);
+} UA_NotificationCallback;
```

Comment:
NotificationCallback has an ambiguous semantics.
You want to callback into userspace. Not the internal notificaiton/subscription layer.